Port "hpx" builds, but does not recognize being on a debug build when being consumed using the vcpk toolchain file and Visual Studio

**Describe the bug**
Port "hpx" builds, but does not recognize being on a debug build when being consumed using the vcpk toolchain file and Visual Studio

**Environment**
- OS: Windows 10
- Compiler: The CXX compiler identification is MSVC 19.27.29112.0

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install hpx:x64-windows
2. ./vcpkg integrate install 
3. Minimal Project (all in one directory):
File CMakeLists.txt:
```cmake
cmake_minimum_required(VERSION 3.17)
project(the_project CXX)
# This setting should not be required
# SET( HPX_IGNORE_CMAKE_BUILD_TYPE_COMPATIBILITY ON)
find_package(HPX)
add_executable   (main "main.cpp")
hpx_setup_target (main COMPONENT_DEPENDENCIES iostreams )
```
File main.cpp
```cpp
#include <hpx/hpx_main.hpp>
#include <hpx/iostream.hpp>
int main(){
    hpx::cout << "Hello, world!";    
    return 0;
}

```
File CmakeSettings.json
```json
{
    "configurations": [
        {
            "name": "x64-Debug",
            "generator": "Ninja",
            "configurationType": "Debug",
            "inheritEnvironments": [ "msvc_x64_x64" ]
        },
        {
            "name": "x64-RelWithDebInfo",
            "generator": "Ninja",
            "configurationType": "RelWithDebInfo",
            "inheritEnvironments": [ "msvc_x64_x64" ]
        }
    ]
}
```

4. Open the project in Visual Studio as "Open Folder" cmake project and try to build the debug build.

**Expected behavior**
The debug build should just work.

**Failure logs** 
Snippet from the Visual Studio Output (cmake configure phase)
```txt
1> [CMake] CMake Error at C:/vcpkg/installed/x64-windows/share/hpx/HPX_Message.cmake:51 (message):
1> [CMake]   CMAKE_BUILD_TYPE does not match Release: this project uses 'Debug', HPX
1> [CMake]   uses 'Release'.  Please add -DCMAKE_BUILD_TYPE=Release to your cmake
1> [CMake]   command or add -DHPX_IGNORE_CMAKE_BUILD_TYPE_COMPATIBILITY=ON to ignore
1> [CMake]   this check (not recommended HPX is not ABI compatible between release and
1> [CMake]   debug modes).
```

**Additional context**
None

--- a/ports/hpx/CONTROL
+++ b/ports/hpx/CONTROL
@@ -1,5 +1,6 @@
 Source: hpx
 Version: 1.5.1
+Port-Version: 1
 Build-Depends: hwloc, boost-accumulators, boost-algorithm, boost-asio, boost-bimap, boost-config, boost-context, boost-dynamic-bitset, boost-exception, boost-filesystem, boost-iostreams, boost-lockfree, boost-program-options, boost-range, boost-spirit, boost-system, boost-throw-exception, boost-variant, boost-winapi
 Homepage: https://github.com/STEllAR-GROUP/hpx
 Description: The C++ Standards Library for Concurrency and Parallelism

--- a/ports/hpx/portfile.cmake
+++ b/ports/hpx/portfile.cmake
@@ -42,6 +42,11 @@ foreach(CMAKE_FILE IN LISTS CMAKE_FILES)
 endforeach()
 
 vcpkg_replace_string(
+    "${CURRENT_PACKAGES_DIR}/share/${PORT}/HPXConfig.cmake"
+    "set(HPX_BUILD_TYPE \"Release\")"
+    "set(HPX_BUILD_TYPE \"\${CMAKE_BUILD_TYPE}\")")
+
+vcpkg_replace_string(
     "${CURRENT_PACKAGES_DIR}/share/${PORT}/HPXMacros.cmake"
     "set(CMAKE_MODULE_PATH \${CMAKE_MODULE_PATH} \"\${CMAKE_CURRENT_LIST_DIR}\")"
     "list(APPEND CMAKE_MODULE_PATH \"\${CMAKE_CURRENT_LIST_DIR}\")")

